hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/QuantileSketch.java

```diff
@@ -38,38 +45,39 @@ public QuantileSketch(final RecordOrder sortOrder, final int resolution) {
     /**
      * Given a table and a desired resolution for percentiles, return the answer for a sample.
      * The size of the sample is resolution*perBin, perBin is set to 100 by default.
+     * The size of the Quantile table we return is slack*resolution. Since this table is shipped
+     * around, we might want to take perBin >> slack to improve the quality of our sample.
      * @param data The input data on which we want to compute Quantiles.
      * @return A table of size resolution, whose ith entry is ranked approximately i/resolution.
      */
     public QuantileList getQuantile(final Table data) {
-        /* Sample a set of rows from the table. */
-        final int perBin = 100;
-        final int dataSize = data.getNumOfRows();
-        /* Sample, then sort the sampled rows. */
-        final IMembershipSet sampleSet = data.members.sample(this.resolution * perBin);
+        /* Sample a set of rows from the table, then sort the sampled rows. */
+        final IMembershipSet sampleSet = data.members.sample(this.resolution * this.perBin);
         final Table sampleTable = data.compress(sampleSet);
         final Integer[] order = this.colSortOrder.getSortedRowOrder(sampleTable);
+        /* We will shrink the set of samples  down to slack*resolution. Number of samples might be
+            less than resolution*perBin, because of repetitions. */
+        final int newRes = Math.min(this.slack * this.resolution, sampleSet.getSize());
+        final int[] quantile = new int[newRes];
+        final WinsAndLosses[] winsAndLosses = new WinsAndLosses[newRes];
+        final double sampleStep = ((double) sampleSet.getSize()+1)/(newRes+ 1);
```

Comment:
The spacing around + is inconsistent; you have all four variations here.
Also, you should not have a space after the cast to (double).